### PR TITLE
Disallow private accessors for virtual properties in interfaces.

### PR DESCRIPTION
**Pull request by AlekseyTs**
@dotnet/roslyn-compiler Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 1]
```diff
@@ -1961,7 +1961,8 @@ class Test1 : I1
                                                          parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
             Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
 
-            // PROTOTYPE(DefaultInterfaceImplementation): We might want to allow code like this.
+            // According to LDM decision captured at https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-04-18.md,
+            // we don't want to allow only one accessor to have an imlementation.
```
Comment (cston) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: Typo, `implementation`, here and other instances below.

File: src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs [hunk 2]
```diff
@@ -233,6 +233,12 @@ public void BinaryOperatorOverloadResolution(BinaryOperatorKind kind, BoundExpre
 
                 foreach (NamedTypeSymbol @interface in interfaces)
                 {
+                    if (!@interface.IsInterface)
+                    {
+                        // this code could be reachable in error situations
+                        continue;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs]: Are we hitting these cases in existing tests?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs]: No, we are not. But we are checking for this case in other similar situations, for example in TypeSymbolExtensions.ImplementsInterface. I am sure we can hit them here, but I don't think it is worth coming up with a test scenario that would have an observable effect in behavior.

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: > Typo, implementation, here and other instances below.

Fixed.

**Comment by cston** [comment]
LGTM

**Comment by AlekseyTs** [comment]
https://ci.dot.net/job/dotnet_roslyn/job/features_DefaultInterfaceImplementation/job/windows_debug_unit64_prtest/48/

xUnit returned error code without errors

xunit produced no error output but had exit code 1
Indication 1

**Comment by AlekseyTs** [comment]
Test windows_debug_unit64_prtest please.

**Pull request merged**